test/errors.spec.js

```diff
@@ -0,0 +1,22 @@
+const expect = require('chai').expect;
+const errors = require('./../src/errors');
+
+const errorNames = Object.keys(errors);
+
+describe('Testing Chester Errors', () => {
+  it('Default Constructor', () => {
+    errorNames.forEach((errorName) => {
+      const error = new errors[errorName]();
+      expect(error).to.have.property('message').and.equal(errorName);
+      expect(JSON.stringify(error)).to.equal(`{"message":"${errorName}"}`);
+    });
+  });
+
+  it('Message Passed to Constructor', () => {
+    errorNames.forEach((errorName) => {
+      const error = new errors[errorName]("It's by design!");
+      expect(error).to.have.property('message').and.equal("It's by design!");
+      expect(JSON.stringify(error)).to.equal('{"message":"It\'s by design!"}');
+    });
```

Comment:
I would move the check out into the `describe` and reuse it in the tests. I.e. 

```js
const validateError = (error, expectedMessage) => {
      expect(error).to.have.property('message').and.equal(expectedMessage);
      expect(JSON.stringify(error)).to.equal(`{"message":"${expectedMessage}"}`);
}
```

(untested)

Reply:
No, it doesn't work. Error native class detects if no arguments passed or `undefined` passed - and behave differently.

Reply:
You'd still do the error initialization inside the test. Only the error validation would be generified. Or am I misunderstanding your point?

Reply:
I tried to instantiate error in that function too.
But without it only a few lines of code will be saved - will it be more readable?
Tests better to keep as plain as possible.

Reply:
Ok, did it, it's your code)